### PR TITLE
Add fr.free.qccrypt.Qccrypt

### PR DESCRIPTION
**Pull request by BofilFR**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by BofilFR** [comment]
Flatpak package for Qccrypt: fr.free.Qccrypt

**Comment by BofilFR** [comment]
I'm not sure I made the things right. I want to publish my app Qccrypt on flathub with the name fr.free.Qccrypt. Please tell me if I made any mistake in the process. Thanks.

File: fr.free.Qccrypt.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      - type: file
+        path: qccrypt-32-x32.png
+      - type: file
+        path: qccrypt-48-x48.png
+      - type: file
+        path: qccrypt-64-x64.png
+      - type: file
+        path: qccrypt-96-x96.png
+      - type: file
+        path: qccrypt-128-x128.png
```
Comment (hfiguiere) [inline fr.free.Qccrypt.yml:44]: all these icon size are unecessary. only 256x256 is useful here.

File: ccrypt-1.11.tar.xz
Comment (hfiguiere) [inline ccrypt-1.11.tar.xz:1]: why is this here?

The flathub repository does not host the upstream sources. Never.

File: qccrypt
Comment (hfiguiere) [inline qccrypt:1]: why is this binary here?

File: fr.free.Qccrypt.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+app-id: fr.free.Qccrypt
+runtime: org.kde.Platform
+runtime-version: 5.15-22.08
+sdk: org.kde.Sdk
+command: qccrypt
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline fr.free.Qccrypt.yml:9]: if it supports wayland then it's `fallback-x11` on line 8

File: fr.free.Qccrypt.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  <id>fr.free.Qccrypt</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0+</project_license>
+  <name>Qccrypt</name>
+  <summary>QT based graphical frontend for ccrypt tool</summary>
+  <developer_name>Philippe Beaureilles</developer_name>
+  <description>
+    <p>QT based graphical frontend for ccrypt tool.</p>
+  </description>
+  <launchable type="desktop-id">fr.free.Qccrypt</launchable>
```
Comment (hfiguiere) [inline fr.free.Qccrypt.appdata.xml:13]: this is an invalid launchable. it should be the file name of the .desktop file.

File: fr.free.Qccrypt.yml (hunk 1 above)
Comment (BofilFR) [reply fr.free.Qccrypt.yml:44]: Thank you for your help. I will remove them and keep only 256x256.

File: ccrypt-1.11.tar.xz
Comment (BofilFR) [reply ccrypt-1.11.tar.xz:1]: Thank you for your help. I will point to the ccrypt website' source code.

File: qccrypt
Comment (BofilFR) [reply qccrypt:1]: I directly put the binary because it fails when I try to compile with qmake within the flatpak package. I will investigate to find a solution and use directly the source code from my web site qccrypt.free.fr.

File: fr.free.Qccrypt.yml (hunk 2 above)
Comment (BofilFR) [reply fr.free.Qccrypt.yml:9]: Thank you for your help. To be honest, I reused a YML file from another flathub project, and it was written like this. So, if I understand, I have to remove the line #8 as my application supports Wayland, that's right?

File: fr.free.Qccrypt.appdata.xml (hunk 3 above)
Comment (BofilFR) [reply fr.free.Qccrypt.appdata.xml:13]: Thank you for your help, I will change it. I'm running out of time right now, so I will make all changes you mentioned in the next few days. Thanks again for your great help!

File: fr.free.Qccrypt.yml (hunk 2 above)
Comment (BofilFR) [reply fr.free.Qccrypt.yml:9]: Ok, sorry, I read wrong. So, if I understand, I have to replace the line `` - --socket=x11`` with `` - --socket=fallback-x11``, that's right?

**Comment by BofilFR** [comment]
Hello. I'm a newbie with github. I made the modifications locally and everything is running fine. How can I push the modifications? I cannot remove the files I uploaded. Thanks in advance for your help.

**Comment by hfiguiere** [comment]
`git push` to the same branch will update the PR automatically.

**Comment by BofilFR** [comment]
It did not work. I need help again. Please find below the commands I entered:

git_push

**Comment by hfiguiere** [comment]
Looks like the origin it uses to push isn't the github repo.

File: fr.free.Qccrypt.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  - name: qccrypt
+    buildsystem: qmake
+    sources:
+      - type: archive
+        url: http://qccrypt.free.fr/download/qccrypt-0.9.1-src.zip
+        sha256: 58b56e7dcf2219c08044d7e9ae1d41144452d178bda30040c707e4dcd290e081
+      - type: file
+        path: fr.free.Qccrypt.desktop
+      - type: file
+        path: qccrypt-256-x256.png
```
Comment (hfiguiere) [inline fr.free.Qccrypt.yml:22]: this is in `qccrypt-0.9.1-src/ressources/images/qccrypt.png`. So it has no need to be in this repository.

File: fr.free.Qccrypt.desktop
Comment (hfiguiere) [inline fr.free.Qccrypt.desktop:1]: this file should in the distribution tarball in a future release, and installed by the build system

File: fr.free.Qccrypt.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+app-id: fr.free.Qccrypt
```
Comment (hfiguiere) [inline fr.free.Qccrypt.yml:1]: because of the way `free.fr` works, a proper app id should be `fr.free.qccrypt.Qccrypt`. This is not dissimilar to code hosting. Upstream controls `gccrypt.free.fr` but not `free.fr`

File: fr.free.Qccrypt.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    sources:
+      - type: archive
+        url: http://qccrypt.free.fr/download/qccrypt-0.9.1-src.zip
+        sha256: 58b56e7dcf2219c08044d7e9ae1d41144452d178bda30040c707e4dcd290e081
+      - type: file
+        path: fr.free.Qccrypt.desktop
+      - type: file
+        path: qccrypt-256-x256.png
+    build-commands:
+      - install -D qccrypt /app/bin/qccrypt
```
Comment (hfiguiere) [inline fr.free.Qccrypt.yml:24]: qmake does't even install the executable?

File: fr.free.Qccrypt.appdata.xml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+<!-- Copyright 2023 Philippe Beaureilles -->
+<component type="desktop-application">
+  <id>fr.free.Qccrypt</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0+</project_license>
+  <name>Qccrypt</name>
+  <summary>QT based graphical frontend for ccrypt tool</summary>
+  <developer_name>Philippe Beaureilles</developer_name>
+  <description>
+    <p>QT based graphical frontend for ccrypt tool.</p>
```
Comment (hfiguiere) [inline fr.free.Qccrypt.appdata.xml:11]: this should also explain what ccrypt is.

File: fr.free.Qccrypt.appdata.xml [hunk 8]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2023 Philippe Beaureilles -->
+<component type="desktop-application">
+  <id>fr.free.Qccrypt</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0+</project_license>
```
Comment (hfiguiere) [inline fr.free.Qccrypt.appdata.xml:6]: This is not a valid spdx license ID.

https://spdx.dev/licenses/

**Comment by hfiguiere** [comment]
None of the question in the submission template have been acknoweldged

**Comment by hfiguiere** [comment]
bot, build fr.free.Qccrypt

File: fr.free.Qccrypt.appdata.xml
[the remark this answers is not in the document]
Comment (hfiguiere) [reply fr.free.Qccrypt.appdata.xml:1]: this is the reaso the build failed right now.

File: fr.free.Qccrypt.yml (hunk 5 above)
Comment (BofilFR) [reply fr.free.Qccrypt.yml:1]: Ok, I will change the name to fr.free.qccrypt.Qccrypt.

File: fr.free.Qccrypt.yml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    buildsystem: qmake
+    sources:
+      - type: archive
+        url: http://qccrypt.free.fr/download/qccrypt-0.9.1-src.zip
+        sha256: 58b56e7dcf2219c08044d7e9ae1d41144452d178bda30040c707e4dcd290e081
+      - type: file
+        path: fr.free.Qccrypt.desktop
+      - type: file
+        path: qccrypt-256-x256.png
+    build-commands:
```
[the remark this answers is not in the document]
Comment (BofilFR) [reply fr.free.Qccrypt.yml:23]: Ok, I will try.

File: fr.free.Qccrypt.appdata.xml (hunk 7 above)
Comment (BofilFR) [reply fr.free.Qccrypt.appdata.xml:11]: Ok, I will add a description for ccrypt.

File: fr.free.Qccrypt.appdata.xml (hunk 8 above)
Comment (BofilFR) [reply fr.free.Qccrypt.appdata.xml:6]: Ok, I will change it.

File: fr.free.Qccrypt.yml (hunk 4 above)
Comment (BofilFR) [reply fr.free.Qccrypt.yml:22]: Humm, ok I didn't know that I could use an image from the source code. I will try to make it work.

File: fr.free.Qccrypt.desktop
Comment (BofilFR) [reply fr.free.Qccrypt.desktop:1]: I'm sorry, I don't understand: how the application's shortcut will be installed if I remove it?

File: fr.free.Qccrypt.yml [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+sdk: org.kde.Sdk
+command: qccrypt
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --filesystem=host
+  - --device=dri
+modules:
+  - name: qccrypt
```
[the remark this answers is not in the document]
Comment (BofilFR) [reply fr.free.Qccrypt.yml:13]: Ok, I will change it. Lot of stuff to do with this flatpak package, I thought it was the last modifications before the validation.

File: fr.free.Qccrypt.desktop
Comment (hfiguiere) [reply fr.free.Qccrypt.desktop:1]: because the build system wold install it like any other package in the last 20 years should. I know it's possible to do that with qmake.

It's a bug to doesn't do that.

**Comment by BofilFR** [comment]
Please check the new version of the .yml file and if it's fine for you, please add fr.free.qqcrypt.Qccrypt. Thanks in advance.

**Comment by hfiguiere** [comment]
bot, build fr.free.qccrypt.Qccrypt
